CMake/Findmlpack.cmake

```diff
@@ -0,0 +1,641 @@
+        )
+      if(_ARMA_USE_${pkg})
+        find_package(${pkg} QUIET)
+        list(APPEND _ARMA_REQUIRED_VARS "${pkg}_FOUND")
+        if(${pkg}_FOUND)
+          list(APPEND _ARMA_SUPPORT_LIBRARIES ${${pkg}_LIBRARIES})
+        endif()
+      endif()
+    endforeach()
+  endif()
+  if (ARMADILLO_FOUND)
+    set(ARMADILLO_INCLUDE_DIRS ${ARMADILLO_INCLUDE_DIR})
+    set(ARMADILLO_LIBRARIES ${ARMADILLO_LIBRARY} ${_ARMA_SUPPORT_LIBRARIES})
+  endif()
+  # Clean up internal variables
+  unset(_ARMA_REQUIRED_VARS)
+  unset(_ARMA_SUPPORT_LIBRARIES)
+  unset(_ARMA_USE_WRAPPER)
+  unset(_ARMA_USE_LAPACK)
+  unset(_ARMA_USE_BLAS)
+  unset(_ARMA_USE_ARPACK)
+  unset(_ARMA_USE_HDF5)
+  unset(_ARMA_CONFIG_CONTENTS)
+  unset(_ARMA_HEADER_CONTENTS)
+
+  cmake_policy(POP)
+endmacro()
+
+# Findcereal.cmake
+macro(find_cereal)
+
+  set(CURRENT_PATH ${ARGN})
+  if (CURRENT_PATH)
+    find_path(CEREAL_INCLUDE_DIR
+      NAMES cereal
+      PATHS "${CURRENT_PATH}/deps/cereal/include"
+      NO_DEFAULT_PATH)
+  else()
+    find_path(CEREAL_INCLUDE_DIR
+      NAMES cereal
+      PATHS "$ENV{ProgramFiles}/cereal/include")
+  endif()
+
+  if (CEREAL_INCLUDE_DIR)
+    # ------------------------------------------------------------------------
+    #  Extract version information from <CEREAL>
+    # ------------------------------------------------------------------------
+    set(CEREAL_FOUND YES)
+    set(CEREAL_VERSION_MAJOR 0)
+    set(CEREAL_VERSION_MINOR 0)
+    set(CEREAL_VERSION_PATCH 0)
+
+    if (EXISTS "${CEREAL_INCLUDE_DIR}/cereal/version.hpp")
+
+      # Read and parse cereal version header file for version number
+      file(READ "${CEREAL_INCLUDE_DIR}/cereal/version.hpp"
+          _CEREAL_HEADER_CONTENTS)
+      string(REGEX REPLACE ".*#define CEREAL_VERSION_MAJOR ([0-9]+).*" "\\1"
+          CEREAL_VERSION_MAJOR "${_CEREAL_HEADER_CONTENTS}")
+      string(REGEX REPLACE ".*#define CEREAL_VERSION_MINOR ([0-9]+).*" "\\1"
+          CEREAL_VERSION_MINOR "${_CEREAL_HEADER_CONTENTS}")
+      string(REGEX REPLACE ".*#define CEREAL_VERSION_PATCH ([0-9]+).*" "\\1"
+          CEREAL_VERSION_PATCH "${_CEREAL_HEADER_CONTENTS}")
+
+    elseif (EXISTS "${CEREAL_INCLUDE_DIR}/cereal/details/polymorphic_impl_fwd.hpp")
+
+      set(CEREAL_VERSION_MAJOR 1)
+      set(CEREAL_VERSION_MINOR 2)
+      set(CEREAL_VERSION_PATCH 0)
+    elseif (EXISTS "${CEREAL_INCLUDE_DIR}/cereal/types/valarray.hpp")
+
+      set(CEREAL_VERSION_MAJOR 1)
+      set(CEREAL_VERSION_MINOR 1)
+      set(CEREAL_VERSION_PATCH 2)
+    elseif (EXISTS "${CEREAL_INCLUDE_DIR}/cereal/cereal.hpp")
+
+    set(CEREAL_VERSION_MAJOR 1)
+    set(CEREAL_VERSION_MINOR 1)
+    set(CEREAL_VERSION_PATCH 1)
+  else()
+
+    set(CEREAL_FOUND NO)
+    endif()
+    set(CEREAL_VERSION_STRING "${CEREAL_VERSION_MAJOR}.${CEREAL_VERSION_MINOR}.${CEREAL_VERSION_PATCH}")
+  endif ()
+
+  mark_as_advanced(CEREAL_INCLUDE_DIR)
+
+  file(GLOB MLPACK_SEARCH_PATHS
+      ${CMAKE_BINARY_DIR}/deps/mlpack-[0-9]*.[0-9]*.[0-9]*)
```

Comment:
Should this line be here?  This is the `find_cereal()` macro, it seems like it shouldn't have anything to do with mlpack.